public/app/features/search/components/SectionHeader.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, useCallback } from 'react';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { Icon, IconButton, IconName, stylesFactory, useTheme } from '@grafana/ui';
+import { DashboardSection, OnToggleChecked } from '../types';
+import appEvents from 'app/core/app_events';
+import { CoreEvents } from 'app/types';
+import { SearchCheckbox } from './SearchCheckbox';
+
+interface SectionHeaderProps {
+  editable?: boolean;
+  onSectionClick: (section: DashboardSection) => void;
+  onToggleChecked?: OnToggleChecked;
+  section: DashboardSection;
+}
+
+export const SectionHeader: FC<SectionHeaderProps> = ({
+  section,
+  onSectionClick,
+  onToggleChecked,
+  editable = false,
+}) => {
+  const theme = useTheme();
+  const styles = getSectionHeaderStyles(theme, section.selected);
+
+  const onSectionExpand = () => {
+    onSectionClick(section);
+  };
+
+  const onSectionChecked = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      e.stopPropagation();
+      if (onToggleChecked) {
+        onToggleChecked(section);
+      }
+    },
+    [section]
+  );
+
+  return !section.hideHeader ? (
+    <div className={styles.wrapper} onClick={onSectionExpand}>
+      <SearchCheckbox editable={editable} checked={section.checked} onClick={onSectionChecked} />
+      <Icon className={styles.icon} name={section.icon as IconName} />
+
+      <span className={styles.text}>{section.title}</span>
+      {section.url && (
+        <a
+          href={section.url}
+          className={styles.link}
+          onClick={() => appEvents.emit(CoreEvents.hideDashSearch, { target: 'search-item' })}
```

Comment:
what is this event for? is this needed / used now that we use a url query parm?

Reply:
Good catch, missed that one.